tools/data-conversion/src/main.rs

```diff
@@ -245,7 +286,10 @@ fn float_to_binary(
 /// # Error
 ///
 /// This function will panic if the input string cannot be parsed as a hexadecimal number.
-fn hex_to_binary(hex_string: &str, filepath_send: &mut File) -> io::Result<()> {
+fn hex_to_binary(
+    hex_string: &str,
+    filepath_send: &mut Option<File>,
```

Comment:
Seems like the `&mut Option<File>` pattern is used to indicate using either STDOUT or a specific file. It might be better to wrap this using something like [`OutputFile`](https://docs.rs/calyx-utils/latest/calyx_utils/enum.OutputFile.html).